java/client/src/main/java/glide/api/models/commands/RangeOptions.java

```diff
@@ -0,0 +1,257 @@
+     */
+    @Getter
+    public static class RangeByLex implements RangeQuery {
+        /** The start lexicographic boundary. */
+        private final String start;
+
+        /** The stop lexicographic boundary. */
+        private final String end;
+
+        /**
+         * The limit argument for a range query. Defaults to null. See <code>Limit</code> class for more
+         * information.
+         */
+        private final Limit limit;
+
+        public RangeByLex(
+                @NonNull RangeOptions.LexRange start,
+                @NonNull RangeOptions.LexRange end,
+                @NonNull Limit limit) {
+            this.start = start.toArgs();
+            this.end = end.toArgs();
+            this.limit = limit;
+        }
+
+        public RangeByLex(@NonNull RangeOptions.LexRange start, @NonNull RangeOptions.LexRange end) {
+            this.start = start.toArgs();
+            this.end = end.toArgs();
+            this.limit = null;
+        }
+    }
+
+    public interface ScoredRangeQuery extends RangeQuery {}
+
+    /**
+     * Represents a range by index (rank) in a sorted set.<br>
+     * The <code>start</code> and <code>stop</code> arguments represent zero-based indexes.
+     */
+    @RequiredArgsConstructor
+    @Getter
+    public static class RangeByIndex implements ScoredRangeQuery {
+        /** The start index of the range. */
+        private final String start;
+
+        /** The stop index of the range. */
+        private final String end;
+
+        public RangeByIndex(long start, long end) {
+            this.start = Long.toString(start);
+            this.end = Long.toString(end);
+        }
+
+        @Override
+        public Limit getLimit() {
+            return null;
+        }
+    }
+
+    /**
+     * Represents a range by score in a sorted set.<br>
+     * The <code>start</code> and <code>stop</code> arguments represent score boundaries.
+     */
+    @Getter
+    public static class RangeByScore implements ScoredRangeQuery {
+        /** The start score boundary. */
+        private final String start;
+
+        /** The stop score boundary. */
+        private final String end;
+
+        /**
+         * The limit argument for a range query. Defaults to null. See <code>Limit</code> class for more
+         * information.
+         */
+        private final Limit limit;
+
+        public RangeByScore(
+                @NonNull RangeOptions.ScoreRange start,
+                @NonNull RangeOptions.ScoreRange end,
+                @NonNull Limit limit) {
+            this.start = start.toArgs();
+            this.end = end.toArgs();
+            this.limit = limit;
+        }
+
+        public RangeByScore(
+                @NonNull RangeOptions.ScoreRange start, @NonNull RangeOptions.ScoreRange end) {
+            this.start = start.toArgs();
+            this.end = end.toArgs();
+            this.limit = null;
+        }
+    }
+
+    public static String[] createZrangeArgs(
```

Comment:
make this non-static, so each kind of query would convert it to args (`String[]`) - like we already have for other things

Reply:
How would you be able to access it without making it static the outer class isn't static.

Reply:
So it would be an instance member, see setoptions for example

Reply:
I don't get the argument we don't even have instance variables in my setup.